packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -87,8 +87,10 @@ function UncontrolledInnerBlocks( props ) {
 
 			const blockType = getBlockType( block.name );
 
-			if ( ! blockType || ! blockType.providesContext ) {
-				return {};
+			if (
+				Object.keys( blockType?.providesContext ?? {} ).length === 0
+			) {
+				return { name: block.name };
```

Comment:
The `name` is used later to get the block layout. Since early return wasn't working, not returning it was never an issue.